torch/csrc/jit/passes/quantization.cpp

```diff
@@ -18,8 +18,30 @@ void PropagateQuantInfo(std::shared_ptr<Graph>& graph) {
   throw std::runtime_error("Pass not implemented yet!");
 }
 
+static void addObserverFor(torch::jit::Value* v) {
+  torch::jit::Node* def = v->node();
+  torch::jit::Node* observer =
+      def->owningGraph()->create(at::Symbol::fromQualString("aten::observer"));
+  def->owningBlock()->appendNode(observer);
+  observer->moveAfter(def);
+  v->replaceAllUsesWith(observer->outputs()[0]);
+  observer->outputs()[0]->setUniqueName(v->uniqueName() + ".observed");
+  observer->addInput(v);
+}
+
 void InsertObserverNodes(std::shared_ptr<Graph>& graph) {
-  throw std::runtime_error("Pass not implemented yet!");
+  for (const auto& n : graph->nodes()) {
+    // Skip nodes that we've just added
+    if (n->kind().toQualString() == std::string("aten::observer")) {
```

Comment:
can you add aten::observer to the interned_strings.h header so we don't reparse it everytime?

Reply:
I would leave that till we actually implement the observer op. E.g. I don't know for sure at the moment whether it will be `aten::` or something else.